From 543bc87eb6ee3ca9e884899f366d007afa01b63d Mon Sep 17 00:00:00 2001
From: Ray Speth <speth@mit.edu>
Date: Fri, 18 Aug 2023 13:23:24 -0400
Subject: [PATCH] [Input] Search active conda environment for YAML files

---
 src/base/application.cpp | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/src/base/application.cpp b/src/base/application.cpp
index ea06488d9..932942c53 100644
--- a/src/base/application.cpp
+++ b/src/base/application.cpp
@@ -224,12 +224,14 @@ void Application::setDefaultDirectories()
     // environment variable).
 #ifdef _WIN32
     string pathsep = ";";
+    string dirsep = "\\";
 #else
     string pathsep = ":";
+    string dirsep = "/";
 #endif
 
-    if (getenv("CANTERA_DATA") != 0) {
-        string s = string(getenv("CANTERA_DATA"));
+    if (getenv("CANTERA_DATA") != nullptr) {
+        string s(getenv("CANTERA_DATA"));
         size_t start = 0;
         size_t end = s.find(pathsep);
         while (end != npos) {
@@ -240,6 +242,13 @@ void Application::setDefaultDirectories()
         inputDirs.push_back(s.substr(start,end));
     }
 
+    // If a conda environment is active, add the location of the Cantera data directory
+    // that may exist in that environment.
+    if (getenv("CONDA_PREFIX") != nullptr) {
+        string s(getenv("CONDA_PREFIX"));
+        inputDirs.push_back(s + dirsep + "share" + dirsep + "cantera" + dirsep + "data");
+    }
+
 #ifdef _WIN32
     // Under Windows, the Cantera setup utility records the installation
     // directory in the registry. Data files are stored in the 'data'
-- 
2.39.2 (Apple Git-143)

