From be146c817ca1f0cfb2898ba32980df2b1203f122 Mon Sep 17 00:00:00 2001
From: Bryan Weber <bryan.w.weber@gmail.com>
Date: Mon, 24 Jun 2024 11:49:19 -0400
Subject: [PATCH 1/2] Cross compiling support is only available without TryRun

---
 SConstruct | 226 ++---------------------------------------------------
 1 file changed, 7 insertions(+), 219 deletions(-)

diff --git a/SConstruct b/SConstruct
index 25483148d..efcea8f54 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1139,74 +1139,16 @@ def get_expression_value(includes, expression, defines=()):
               '}\n'))
     return '\n'.join(s)

-# Check that libraries link correctly
-cmath_check_source = get_expression_value(["<cmath>"], "cos(0. * argc)")
-retcode, cmath_works = conf.TryRun(cmath_check_source, ".cpp")
-if cmath_works.strip() != "1":
-    config_error(
-        "The C++ compiler is not correctly configured (failed at linking stage)."
-    )
-
-# Check that NaN is treated correctly
-nan_check_source = get_expression_value(["<cmath>"], "std::isnan(NAN + argc)")
-retcode, nan_works = conf.TryRun(nan_check_source, ".cpp")
-if nan_works.strip() != "1":
-    config_error(
-        "Cantera requires a working implementation of 'std::isnan'.\n"
-        "If you have specified '-ffast-math' or equivalent as an optimization option,\n"
-        "either remove this option or add the '-fno-finite-math-only option'."
-    )
-
-def split_version(version):
-    """Split integer version into version string."""
-    version = divmod(float(version.strip()), 10000)
-    (fmt_maj, (fmt_min, fmt_pat)) = version[0], divmod(version[1], 100)
-    return f"{fmt_maj:.0f}.{fmt_min:.0f}.{fmt_pat:.0f}"
-
 # Check for fmt library and checkout submodule if needed
 if env['system_fmt'] in ('y', 'default'):
-    fmt_version_source = get_expression_value(
-        ['<fmt/format.h>'], 'FMT_VERSION', ['FMT_HEADER_ONLY'])
-    retcode, fmt_lib_version = conf.TryRun(fmt_version_source, '.cpp')
-    if retcode:
-        fmt_lib_version = split_version(fmt_lib_version)
-        fmt_min_version = "6.1.2"
-        if parse_version(fmt_lib_version) < parse_version(fmt_min_version):
-            if env['system_fmt'] == 'y':
-                config_error(
-                    f"System fmt version {fmt_lib_version} is not supported;"
-                    f"version {fmt_min_version} or higher is required.")
-                logger.info(
-                    f"System fmt version {fmt_lib_version} is not supported; "
-                    "using private installation instead.")
-        else:
-            env['system_fmt'] = True
-            logger.info(f"Using system installation of fmt library.")
-    elif env['system_fmt'] == 'y':
-        config_error('Expected system installation of fmt library, but it '
-            'could not be found.')
+    if conf.CheckCXXHeader("fmt/ostream.h", '""'):
+        env["system_fmt"] = True
+        logger.info("Using system installation of fmt library")

 if env['system_fmt'] in ('n', 'default'):
     if not os.path.exists('ext/fmt/include/fmt/ostream.h'):
         checkout_submodule("fmt", "ext/fmt")

-    fmt_version_source = get_expression_value(
-        ['"../ext/fmt/include/fmt/format.h"'], 'FMT_VERSION', ['FMT_HEADER_ONLY'])
-    retcode, fmt_lib_version = conf.TryRun(fmt_version_source, '.cpp')
-    if not retcode:
-        config_error('Expected private installation of fmt library, but it is '
-            'not configured correctly.')
-
-    fmt_lib_version = split_version(fmt_lib_version)
-    env['system_fmt'] = False
-    logger.info(f"Using private installation of fmt library.")
-
-if env["OS"] == "Windows" and parse_version(fmt_lib_version) < parse_version("8.0.0"):
-    # Workaround for symbols not exported on Windows in older fmt versions
-    env.Append(CPPDEFINES={"FMT_HEADER_ONLY": 1})
-
-logger.info(f"Using fmt version {fmt_lib_version}")
-
 # Check for yaml-cpp library and checkout submodule if needed
 if env['system_yamlcpp'] in ('y', 'default'):
     # We need the Mark() function, which was added in version 0.5.3
@@ -1268,12 +1210,6 @@ if env["system_eigen"] in ("n", "default"):
         checkout_submodule("Eigen", "ext/eigen")
     eigen_include = '"../ext/eigen/Eigen/Core"'

-eigen_versions = 'QUOTE(EIGEN_WORLD_VERSION) "." QUOTE(EIGEN_MAJOR_VERSION) "." QUOTE(EIGEN_MINOR_VERSION)'
-eigen_version_source = get_expression_value([eigen_include], eigen_versions)
-retcode, eigen_lib_version = conf.TryRun(eigen_version_source, ".cpp")
-env["EIGEN_LIB_VERSION"] = eigen_lib_version.strip()
-logger.info(f"Found Eigen version {env['EIGEN_LIB_VERSION']}")
-
 # Determine which standard library to link to when using Fortran to
 # compile code that links to Cantera
 if conf.CheckDeclaration('__GLIBCXX__', '#include <iostream>', 'C++'):
@@ -1296,19 +1232,6 @@ else:
     env["HAS_OPENMP"] = False
     logger.info("Not checking for OpenMP support due to using XCode compiler.")

-boost_version_source = get_expression_value(['<boost/version.hpp>'], 'BOOST_LIB_VERSION')
-retcode, boost_lib_version = conf.TryRun(boost_version_source, '.cpp')
-env['BOOST_LIB_VERSION'] = '.'.join(boost_lib_version.strip().split('_'))
-if not env['BOOST_LIB_VERSION']:
-    config_error("Boost could not be found. Install Boost headers or set"
-                 " 'boost_inc_dir' to point to the boost headers.")
-else:
-    logger.info(f"Found Boost version {env['BOOST_LIB_VERSION']}")
-if parse_version(env['BOOST_LIB_VERSION']) < parse_version("1.70"):
-    # Boost.DLL with std::filesystem (making it header-only) is available in Boost 1.70
-    # or newer
-    config_error("Cantera requires Boost version 1.70 or newer.")
-
 # check BLAS/LAPACK installations
 if env["system_blas_lapack"] == "n":
     env["blas_lapack_libs"] = []
@@ -1344,32 +1267,6 @@ elif env["system_blas_lapack"] == "default":
     if not env["blas_lapack_libs"] and env["OS"] != "Darwin":
         logger.info("No system BLAS/LAPACK libraries detected.")

-if "mkl_rt" in env["blas_lapack_libs"]:
-    mkl_version = textwrap.dedent("""\
-        #include <iostream>
-        #include "mkl.h"
-        int main(int argc, char** argv) {
-            MKLVersion Version;
-            mkl_get_version(&Version);
-            std::cout << Version.MajorVersion << "." << Version.MinorVersion << "."
-                << Version.UpdateVersion << " for " << Version.Platform;
-            return 0;
-        }
-    """)
-    retcode, mkl_version = conf.TryRun(mkl_version, ".cpp")
-    if retcode:
-        logger.info(f"Using MKL {mkl_version}")
-    else:
-        logger.warning("Failed to determine MKL version.")
-
-elif "openblas" in env["blas_lapack_libs"]:
-    openblas_version_source = get_expression_value(["<openblas_config.h>"], "OPENBLAS_VERSION")
-    retcode, openblas_version = conf.TryRun(openblas_version_source, ".cpp")
-    if retcode:
-        logger.info(f"Using {openblas_version.strip()}")
-    else:
-        logger.warning("Failed to determine OpenBLAS version.")
-
 import SCons.Conftest, SCons.SConf
 context = SCons.SConf.CheckContext(conf)

@@ -1415,24 +1312,8 @@ if env['system_sundials'] == 'y':
                    'sunnonlinsol'):
         remove_directory('include/cantera/ext/' + subdir)

-    # Determine Sundials version
-    sundials_version_source = get_expression_value(['"sundials/sundials_config.h"'],
-                                                   'QUOTE(SUNDIALS_VERSION)')
-    retcode, sundials_version = conf.TryRun(sundials_version_source, '.cpp')
-    if retcode == 0:
-        config_error("Failed to determine Sundials version.")
-    sundials_version = sundials_version.strip(' "\n')
-
-    # Ignore the minor version, for example 2.4.x -> 2.4
-    env['sundials_version'] = '.'.join(sundials_version.split('.')[:2])
+    env['sundials_version'] = "7.0.0"
     sundials_ver = parse_version(env['sundials_version'])
-    if sundials_ver < parse_version("3.0") or sundials_ver >= parse_version("8.0"):
-        logger.error(f"Sundials version {env['sundials_version']!r} is not supported.")
-        sys.exit(1)
-    elif sundials_ver > parse_version("7.0.0"):
-        logger.warning(f"Sundials version {env['sundials_version']!r} has not been tested.")
-
-    logger.info(f"Using system installation of Sundials version {sundials_version!r}.")

     # Determine whether or not Sundials was built with BLAS/LAPACK
     if sundials_ver < parse_version('5.5'):
@@ -1464,102 +1345,10 @@ else: # env['system_sundials'] == 'n'
     env['sundials_version'] = '5.3'
     env['has_sundials_lapack'] = int(env['use_lapack'])

-if env["hdf_include"] and env["hdf_support"] in ("y", "default"):
-    env["hdf_include"] = Path(env["hdf_include"]).as_posix()
-    add_system_include(env, env["hdf_include"])
-    env["hdf_support"] = "y"
-if env["hdf_libdir"] and env["hdf_support"] in ("y", "default"):
-    env["hdf_libdir"] = Path(env["hdf_libdir"]).as_posix()
-    env.Append(LIBPATH=[env["hdf_libdir"]])
-    env["hdf_support"] = "y"
-    if env["use_rpath_linkage"]:
-        env.Append(RPATH=env["hdf_libdir"])
-
-if env["hdf_support"] == "n":
-    env["use_hdf5"] = False
-else:
-    libhdf_exists = conf.CheckLib("hdf5", autoadd=False)
-    libhdf_serial_exists = conf.CheckLib("hdf5_serial", autoadd=False)
-    env["use_hdf5"] = False
-    if libhdf_exists:
-        env["use_hdf5"] = True
-        env["hdf5_lib"] = "hdf5"
-    elif libhdf_serial_exists:
-        env["use_hdf5"] = True
-        env["hdf5_lib"] = "hdf5_serial"
-        if not env["hdf_include"]:
-            add_system_include(env, "/usr/include/hdf5/serial/")
-    if not env["use_hdf5"] and env["hdf_support"] == "y":
-        config_error("HDF5 support has been specified but libraries were not found.")
-
-if env["use_hdf5"] and env["system_highfive"] in ("y", "default"):
-    if conf.CheckLibWithHeader(
-            "hdf5", "highfive/H5File.hpp", language="C++", autoadd=False):
-
-        highfive_include = "<highfive/H5Version.hpp>"
-        h5_version_source = get_expression_value(
-            [highfive_include], "QUOTE(HIGHFIVE_VERSION)")
-        retcode, h5_lib_version = conf.TryRun(h5_version_source, ".cpp")
-        if retcode:
-            if parse_version(h5_lib_version) < parse_version("2.5"):
-                if env["system_highfive"] == "y":
-                    config_error(
-                        f"System HighFive version {h5_lib_version} is not "
-                        "supported; version 2.5 or higher is required.")
-                logger.info(
-                    f"System HighFive version {h5_lib_version} is not supported. "
-                    "Using private installation instead.")
-            else:
-                env["system_highfive"] = True
-                logger.info("Using system installation of HighFive library.")
-            env["HIGHFIVE_VERSION"] = h5_lib_version.strip()
-        else:
-            config_error("Detected invalid HighFive configuration.")
-
-        highfive_include = "<highfive/H5DataType.hpp>"
-
-    elif env["system_highfive"] == "y":
-        config_error(
-            "Expected system installation of HighFive library, but it is either "
-            "corrupted or could not be found.")
-
-if env["use_hdf5"] and env["system_highfive"] in ("n", "default"):
-    env["system_highfive"] = False
-    if not Path("ext/HighFive/include").is_dir():
-        checkout_submodule("HighFive", "ext/HighFive")
-
-    def highfive_version(cmake_lists):
-        """Read highfive version from CMakeLists.txt"""
-        h5_version = Path(cmake_lists).read_text()
-        h5_version = [line for line in h5_version.split("\n")
-                      if line.startswith("project(HighFive")]
-        return re.search(r'[0-9]+\.[0-9]+\.[0-9]+', h5_version[0]).group(0)
-
-    env["HIGHFIVE_VERSION"] = highfive_version("ext/HighFive/CMakeLists.txt")
-    highfive_include = '"../ext/HighFive/include/highfive/H5DataType.hpp"'
-    logger.info(f"Using private installation of HighFive library.")
-
-if env["use_hdf5"]:
-    if conf.CheckStatement("HighFive::details::Boolean::HighFiveTrue",
-                           f"#include {highfive_include}"):
-        env["highfive_boolean"] = True
-
-    hdf_version = textwrap.dedent("""\
-        #include <iostream>
-        #include "H5public.h"
-        int main(int argc, char** argv) {
-            std::cout << H5_VERS_MAJOR << "." << H5_VERS_MINOR << "." << H5_VERS_RELEASE;
-            return 0;
-        }
-    """)
-    retcode, hdf_version = conf.TryRun(hdf_version, ".cpp")
-    if retcode:
-        env["HDF_VERSION"] = hdf_version
-        logger.info(
-            f"Using HighFive version {env['HIGHFIVE_VERSION']} "
-            f"for HDF5 {env['HDF_VERSION']}")
-    else:
-        logger.warning("Failed to determine HDF5 version.")
+env["system_highfive"] = True
+env["use_hdf5"] = True
+env["highfive_boolean"] = True
+env["hdf5_lib"] = "hdf5"

 def set_fortran(pattern, value):
     # Set compiler / flags for all Fortran versions to be the same
--
2.45.2
