From 6a156be543586337e55ae33aa3231c3d5946dab5 Mon Sep 17 00:00:00 2001
From: Bryan Weber <bryan.w.weber@gmail.com>
Date: Mon, 24 Jun 2024 11:51:41 -0400
Subject: [PATCH 2/2] Remove check for iomp5

---
 SConstruct | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/SConstruct b/SConstruct
index efcea8f54..3a994b159 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1220,17 +1220,7 @@ else:
     env['cxx_stdlib'] = []
 
 env['HAS_CLANG'] = conf.CheckDeclaration('__clang__', '', 'C++')
-if not env["using_apple_clang"]:
-    # This checks for these three libraries in order and stops when it finds the
-    # first success. Intel = iomp5, LLVM/clang = omp, GCC = gomp. Since gomp is
-    # likely to be installed on the system even if other compilers are installed
-    # or in use, it needs to go last in the check.
-    env['HAS_OPENMP'] = conf.CheckLibWithHeader(
-        ["iomp5", "omp", "gomp"], "omp.h", language="C++"
-    )
-else:
-    env["HAS_OPENMP"] = False
-    logger.info("Not checking for OpenMP support due to using XCode compiler.")
+env["HAS_OPENMP"] = False
 
 # check BLAS/LAPACK installations
 if env["system_blas_lapack"] == "n":
-- 
2.45.2