From 9013cd47fc7871b9df0e36ed3e20ec102e6065d1 Mon Sep 17 00:00:00 2001
From: Bryan Weber <bryan.w.weber@gmail.com>
Date: Thu, 5 May 2022 17:44:11 -0500
Subject: [PATCH] Remove check for iomp5

---
 SConstruct | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/SConstruct b/SConstruct
index d55f20fb0..f61119aef 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1203,17 +1203,9 @@ else:
     env['cxx_stdlib'] = []
 
 env['HAS_CLANG'] = conf.CheckDeclaration('__clang__', '', 'C++')
-if not env["using_apple_clang"]:
-    # This checks for these three libraries in order and stops when it finds the
-    # first success. Intel = iomp5, LLVM/clang = omp, GCC = gomp. Since gomp is
-    # likely to be installed on the system even if other compilers are installed
-    # or in use, it needs to go last in the check.
-    env['HAS_OPENMP'] = conf.CheckLibWithHeader(
-        ["iomp5", "omp", "gomp"], "omp.h", language="C++"
-    )
-else:
-    env["HAS_OPENMP"] = False
-    logger.info("Not checking for OpenMP support due to using XCode compiler.")
+
+env["HAS_OPENMP"] = False
+
 # demangle is available in Boost 1.56 or newer
 env['has_demangle'] = conf.CheckDeclaration("boost::core::demangle",
                                 '#include <boost/core/demangle.hpp>', 'C++')
-- 
2.36.0

