From 139a3f1cd8bfc49dd192fc9e2ed84a9ed7dff868 Mon Sep 17 00:00:00 2001
From: Bryan Weber <bryan.w.weber@gmail.com>
Date: Thu, 5 May 2022 17:44:11 -0500
Subject: [PATCH] Remove check for iomp5

---
 SConstruct | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/SConstruct b/SConstruct
index c21b2fc14..9cb0ae423 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1238,19 +1238,6 @@ elif conf.CheckDeclaration('_LIBCPP_VERSION', '#include <iostream>', 'C++'):
 else:
     env['cxx_stdlib'] = []
 
-env['HAS_CLANG'] = conf.CheckDeclaration('__clang__', '', 'C++')
-if not env["using_apple_clang"]:
-    # This checks for these three libraries in order and stops when it finds the
-    # first success. Intel = iomp5, LLVM/clang = omp, GCC = gomp. Since gomp is
-    # likely to be installed on the system even if other compilers are installed
-    # or in use, it needs to go last in the check.
-    env['HAS_OPENMP'] = conf.CheckLibWithHeader(
-        ["iomp5", "omp", "gomp"], "omp.h", language="C++"
-    )
-else:
-    env["HAS_OPENMP"] = False
-    logger.info("Not checking for OpenMP support due to using XCode compiler.")
-
 boost_version_source = get_expression_value(['<boost/version.hpp>'], 'BOOST_LIB_VERSION')
 retcode, boost_lib_version = conf.TryRun(boost_version_source, '.cpp')
 env['BOOST_LIB_VERSION'] = '.'.join(boost_lib_version.strip().split('_'))
-- 
2.36.0

