From 6d148584193e623b34a4bd83d793132eb02ba9b2 Mon Sep 17 00:00:00 2001
From: Bryan Weber <bryan.w.weber@gmail.com>
Date: Fri, 21 Jun 2024 20:32:33 -0400
Subject: [PATCH 3/3] Remove check for iomp5

---
 SConstruct | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/SConstruct b/SConstruct
index 829b06e53..179b9d5f1 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1289,17 +1289,7 @@ else:
     env['cxx_stdlib'] = []
 
 env['HAS_CLANG'] = conf.CheckDeclaration('__clang__', '', 'C++')
-if not env["using_apple_clang"]:
-    # This checks for these three libraries in order and stops when it finds the
-    # first success. Intel = iomp5, LLVM/clang = omp, GCC = gomp. Since gomp is
-    # likely to be installed on the system even if other compilers are installed
-    # or in use, it needs to go last in the check.
-    env['HAS_OPENMP'] = conf.CheckLibWithHeader(
-        ["iomp5", "omp", "gomp"], "omp.h", language="C++"
-    )
-else:
-    env["HAS_OPENMP"] = False
-    logger.info("Not checking for OpenMP support due to using XCode compiler.")
+env["HAS_OPENMP"] = False
 
 # check BLAS/LAPACK installations
 if env["system_blas_lapack"] == "n":
-- 
2.45.2

