From 968c9c6118ef533fca25c51e3c1e26c434261d7c Mon Sep 17 00:00:00 2001
From: Ray Speth <speth@mit.edu>
Date: Tue, 15 Aug 2023 18:54:16 -0400
Subject: [PATCH] Remove check for iomp5

---
 SConstruct | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/SConstruct b/SConstruct
index 7588630f3..e7ab6ae1d 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1296,17 +1296,7 @@ else:
     env['cxx_stdlib'] = []
 
 env['HAS_CLANG'] = conf.CheckDeclaration('__clang__', '', 'C++')
-if not env["using_apple_clang"]:
-    # This checks for these three libraries in order and stops when it finds the
-    # first success. Intel = iomp5, LLVM/clang = omp, GCC = gomp. Since gomp is
-    # likely to be installed on the system even if other compilers are installed
-    # or in use, it needs to go last in the check.
-    env['HAS_OPENMP'] = conf.CheckLibWithHeader(
-        ["iomp5", "omp", "gomp"], "omp.h", language="C++"
-    )
-else:
-    env["HAS_OPENMP"] = False
-    logger.info("Not checking for OpenMP support due to using XCode compiler.")
+env['HAS_OPENMP'] = False
 
 # check BLAS/LAPACK installations
 if env["system_blas_lapack"] == "n":
-- 
2.39.2 (Apple Git-143)

