From 2e75a25cb1b77bfe0a76c8cf7ed94243ad13f69a Mon Sep 17 00:00:00 2001
From: Bryan Weber <bryan.w.weber@gmail.com>
Date: Fri, 26 Jul 2024 09:53:03 -0400
Subject: [PATCH 1/1] Link fmt in CMake as an external library

---
 samples/cxx/SConscript | 25 ++++++++++++++++---------
 1 file changed, 16 insertions(+), 9 deletions(-)

diff --git a/samples/cxx/SConscript b/samples/cxx/SConscript
index 63ee37d29..7bc3a453a 100644
--- a/samples/cxx/SConscript
+++ b/samples/cxx/SConscript
@@ -21,22 +21,23 @@ samples = [
 
 for subdir, name, extensions, openmp in samples:
     localenv = env.Clone()
+    cmake_extra = []
     if openmp:
         localenv.Append(CXXFLAGS=env['openmp_flag'], LINKFLAGS=env['openmp_flag'])
         if env['using_apple_clang']:
             localenv.Append(LIBS=['omp'])
 
-        localenv['cmake_extra'] = """
-find_package(OpenMP REQUIRED)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
-"""
-    else:
-        localenv['cmake_extra'] = ''
+        cmake_extra.extend(
+            [
+                "find_package(OpenMP REQUIRED)",
+                'set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")',
+                'set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")',
+            ]
+        )
 
     # TODO: Accelerate is only used if other BLAS/LAPACK are not used
     if env["OS"] == "Darwin":
-        localenv["cmake_extra"] += "find_library(ACCELERATE_FRAMEWORK Accelerate)"
+        cmake_extra.append("find_library(ACCELERATE_FRAMEWORK Accelerate)")
         localenv.Append(
             LINKFLAGS=env.subst("${RPATHPREFIX}${ct_libdir}${RPATHSUFFIX}"))
 
@@ -77,6 +78,11 @@ set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}
         localenv.Append(LINKFLAGS=[env.subst(f'$RPATHPREFIX{d}$RPATHSUFFIX')
                                    for d in libdirs])
 
+    cmake_libs = localenv['cantera_shared_libs'].copy()
+    if "fmt" in cmake_libs:
+        cmake_extra.append("find_package(fmt REQUIRED)")
+        cmake_libs[cmake_libs.index("fmt")] = "fmt::fmt"
+
     cc_flags = compiler_flag_list(localenv["CCFLAGS"] + localenv["CXXFLAGS"],
                                   env["CC"], flag_excludes)
     link_flags = compiler_flag_list(localenv["LINKFLAGS"], env["CC"], flag_excludes)
@@ -85,7 +91,7 @@ set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}
     localenv['tmpl_cantera_incdirs'] = repr([x for x in incdirs if x])
     localenv['cmake_cantera_incdirs'] = ' '.join(quoted(x) for x in incdirs if x)
     localenv['tmpl_cantera_libs'] = repr(localenv['cantera_shared_libs'])
-    localenv['cmake_cantera_libs'] = ' '.join(localenv['cantera_shared_libs'])
+    localenv['cmake_cantera_libs'] = ' '.join(cmake_libs)
     if env['OS'] == 'Darwin':
         localenv['cmake_cantera_libs'] += ' ${ACCELERATE_FRAMEWORK}'
         localenv['cmake_cantera_incdirs'] += ' "/usr/local/include"'
@@ -94,6 +100,7 @@ set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}
     localenv['tmpl_cantera_linkflags'] = repr(link_flags)
     localenv['tmpl_progname'] = name
     localenv['tmpl_sourcename'] = name + '.cpp'
+    localenv['cmake_extra'] = "\n".join(cmake_extra)
     env_args = []
 
     ## Generate SConstruct files to be installed
-- 
2.45.2

